doc/counters_enabling_redesign.md

```diff
@@ -0,0 +1,66 @@
+# Counters Enabling Redesign
+
+## Table of Content
+
+* [Revision](#revision)
+* [Overview](#overview)
+* [Requirements](#requirements)
+* [High-Level Design](#high-level-design)
+    * [Ports Daemon](#ports-daemon)
+
+
+### Revision
+
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |  12.15.2021 | Noa Or             | Initial version                   |
+
+
+## Overview
+
+In the current design of counter enabling, when enable_counters.py script is triggered, if the switch uptime is less then 5 minutes, it sleeps for 3 minutes. This is done in order to make sure the system is up and synchronized.
+
+The purpose of counters redesign is to change the 'sleep' mechanisem to be event driven and to have the exact time when the siwtch is ready to enable counters.
+
+## Requirements
+
+Change enable_counters.py script to be event driven from config_db instead of using sleeps.
+
+
+## High-Level Design
+
+### Ports Daemon
+
+enable_counters.py script will be refactored to be running as a daemon.
```

Comment:
this should be merged into the orchagent

Reply:
please also add config db schame and yang schema for this feature.

Reply:
@lguohan 
Updated the design according to your request. 
regarding the config db schema & yang model - why do we need to add config db schema? we don't add anything new to DB - it's all internal. 
If you mean the delay_status attributes, we already have schema & yang for them. 
https://github.com/Azure/sonic-swss-common/blob/master/common/schema.h#L225
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-flex_counter.yang#L49

Reply:
i think in the config db, we need to add something to set the delay time, for example 300 seconds which allow user to specify how long they want to delay?

Reply:
@lguohan 
This PR is related to the fact we are moving from sleep mechanism to be event driven. 
I agree that adding an option of how much delay the user wants is good, but I don't think it's related to this design. 

If you want,  our team will take this feature, but in another design and plan. 
what do you think?